instrumentation-api/src/test/java/io/opentelemetry/instrumentation/api/config/ConfigTest.java

```diff
@@ -144,17 +150,65 @@ void shouldGetMap() {
             .addProperty("prop.wrong", "one=1, but not two!")
             .build();
 
-    assertEquals(map("one", "1", "two", "2"), config.getMapProperty("prop.map"));
+    assertEquals(map("one", "1", "two", "2"), config.getMap("prop.map"));
     assertEquals(
-        map("one", "1", "two", "2"), config.getMapProperty("prop.map", singletonMap("three", "3")));
-    assertTrue(config.getMapProperty("prop.wrong").isEmpty());
+        map("one", "1", "two", "2"), config.getMap("prop.map", singletonMap("three", "3")));
+    assertTrue(config.getMap("prop.wrong").isEmpty());
     assertEquals(
-        singletonMap("three", "3"),
-        config.getMapProperty("prop.wrong", singletonMap("three", "3")));
-    assertTrue(config.getMapProperty("prop.missing").isEmpty());
+        singletonMap("three", "3"), config.getMap("prop.wrong", singletonMap("three", "3")));
+    assertTrue(config.getMap("prop.missing").isEmpty());
     assertEquals(
-        singletonMap("three", "3"),
-        config.getMapProperty("prop.missing", singletonMap("three", "3")));
+        singletonMap("three", "3"), config.getMap("prop.missing", singletonMap("three", "3")));
+  }
+
+  @ParameterizedTest
+  @MethodSource("agentDebugParams")
+  void shouldCheckIfAgentDebugModeIsEnabled(String propertyValue, boolean expected) {
+    Config config = Config.newBuilder().addProperty("otel.javaagent.debug", propertyValue).build();
+
+    assertEquals(expected, config.isAgentDebugEnabled());
+  }
+
+  @SuppressWarnings("unused")
+  private static Stream<Arguments> agentDebugParams() {
+    return Stream.of(
+        Arguments.of("true", true), Arguments.of("blather", false), Arguments.of(null, false));
+  }
+
+  @ParameterizedTest
+  @MethodSource("instrumentationEnabledParams")
```

Comment:
I tend to prefer `ArgumentsSource` for the compile safety at the expense of some lines of class definition IntelliJ quickly completes. Looks like it would remove the `unused` suppression